Defining a `serverSetup` function without args crashes the server

### Issue
If the user defines the `serverSetupFn` like this:
```js
export default async function () {
  await github.submit()
  await loadTime.submit({
    url: "https://wasp-lang.dev",
    name: "wasp-lang.dev Load Time"
  })
}
```
the server crashes with the error:
```
src/server.ts(21,21): error TS2554: Expected 0 arguments, but got 1.
```

### Possible solution
If the user changed *his* function as:
```js
export default async function (context) {
```
it would work.

This makes the newest addition https://github.com/wasp-lang/wasp/pull/1017 a breaking change.

### Making it backwards compatible

If we changed our `server.ts` from:
```ts
await serverSetup(serverSetupFnContext)
```
to
```ts
await (serverSetup as ServerSetupFn)(serverSetupFnContext)
```
it would work 😃

--- a/waspc/data/Generator/templates/server/src/server.ts
+++ b/waspc/data/Generator/templates/server/src/server.ts
@@ -6,7 +6,7 @@ import config from './config.js'
 
 {=# doesServerSetupFnExist =}
 {=& serverSetupJsFnImportStatement =}
-import { ServerSetupFnContext } from './types'
+import { ServerSetupFn, ServerSetupFnContext } from './types'
 {=/ doesServerSetupFnExist =}
 
 {=# isPgBossJobExecutorUsed =}
@@ -26,7 +26,7 @@ const startServer = async () => {
 
   {=# doesServerSetupFnExist =}
   const serverSetupFnContext: ServerSetupFnContext = { app, server }
-  await {= serverSetupJsFnIdentifier =}(serverSetupFnContext)
+  await ({= serverSetupJsFnIdentifier =} as ServerSetupFn)(serverSetupFnContext)
   {=/ doesServerSetupFnExist =}
 
   server.listen(port)

